packages/cocoon_common/lib/src/time_range.dart

```diff
@@ -0,0 +1,161 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+
+/// A definition of a time range, which can be specific or indefinite.
+@immutable
+sealed class TimeRange {
+  const TimeRange();
+
+  /// Matches any time.
+  static const TimeRange indefinite = IndefiniteTimeRange();
+
+  /// Creates a range matching time between [start] and [end].
+  ///
+  /// If [exclusive] is set, the specific moments described in [start] and [end]
+  /// are excluded.
+  ///
+  /// [end] must be after [start].
+  factory TimeRange.between(
+    DateTime start, //
+    DateTime end, {
+    bool exclusive,
```

Comment:
Is this either: 
`(start, end)` or `[start, end]`
and never `(start, end]` or `[start, end)`